gnocchi/rest/aggregates/api.py

```diff
@@ -238,36 +220,38 @@ def post(self, start=None, stop=None, granularity=None,
 
             groupby = sorted(set(api.arg_to_list(groupby)))
             sorts = groupby if groupby else api.RESOURCE_DEFAULT_PAGINATION
+            results = []
             try:
-                resources = pecan.request.indexer.list_resources(
-                    body["resource_type"],
-                    attribute_filter=attr_filter,
-                    sorts=sorts)
-            except indexer.IndexerException as e:
-                api.abort(400, six.text_type(e))
-            if not groupby:
-                try:
+                if not groupby:
+                    resources = pecan.request.indexer.list_resources(
+                        body["resource_type"],
+                        attribute_filter=attr_filter,
+                        sorts=sorts)
                     return self._get_measures_by_name(
                         resources, references, body["operations"], start, stop,
                         granularity, needed_overlap, fill, details=details)
-                except indexer.NoSuchMetric as e:
-                    api.abort(404, e)
 
-            def groupper(r):
-                return tuple((attr, r[attr]) for attr in groupby)
+                if use_history:
+                    results = self.get_measures_grouping_with_history(
+                        attr_filter, body, details, fill, granularity, groupby,
+                        needed_overlap, references, sorts, start, stop)
+                else:
+                    resources = pecan.request.indexer.list_resources(
+                        body["resource_type"],
+                        attribute_filter=attr_filter,
+                        sorts=sorts)
+                    results = self.get_measures_grouping(
+                        body, details, fill, granularity, needed_overlap,
+                        references, resources, start, stop, groupby)
+
+            except indexer.NoSuchMetric as e:
+                api.abort(404, e)
+            except indexer.IndexerException as e:
+                api.abort(400, six.text_type(e))
+            except Exception as e:
+                LOG.exception(e)
```

Comment:
probably not the right place to log

Reply:
Hmm, maybe, where do you suggest I should put this log? I cant think in another place. Thanks :slightly_smiling_face:

Reply:
See my comment below on the same point.

Reply:
It will not return an empty result, it will raise the Exception again (and maybe return error 500, or 404, it will depends how the caller is handling exceptions, but it is how it was implemented before (except by the log stuff)